lib/compat/wordpress-6.7/blocks.php

```diff
@@ -43,3 +43,25 @@ function gutenberg_filter_block_type_metadata_settings_allow_variations_php_file
 	return $settings;
 }
 add_filter( 'block_type_metadata_settings', 'gutenberg_filter_block_type_metadata_settings_allow_variations_php_file', 10, 2 );
+
+/**
+ * Filters the block type arguments during registration to stabilize experimental block supports.
+ *
+ * This is a temporary compatibility shim as the approach in core is for this to be handled
+ * within the WP_Block_Type class rather than requiring a filter.
+ *
+ * @param array $args Array of arguments for registering a block type.
+ * @return array Array of arguments for registering a block type.
+ */
+function gutenberg_stabilize_experimental_block_supports( $args ) {
+
+	if ( empty( $args['supports']['__experimentalBorder'] ) ) {
+		return $args;
+	}
+
+	$args['supports']['border'] = $args['supports']['__experimentalBorder'];
+
+	return $args;
+}
+
+add_filter( 'register_block_type_args', 'gutenberg_stabilize_experimental_block_supports', PHP_INT_MAX, 1 );
```

Comment:
As this won't land until 6.8, we'll need to move this to the 6.8 compat folder. Given there's already the same filter equivalent for typography supports, we probably only need to move the border stabilization into that same filter function.